invenio_records_resources/services/files/schema.py

```diff
@@ -85,7 +85,11 @@ def fields_from_file_obj(self, data, **kwargs):
         if data.file:
             # mandatory fields
             data["storage_class"] = data.file.storage_class
-            data["uri"] = data.file.uri  # TODO: only serialize for external?
+            data["uri"] = data.file.uri
+
+            # If Local -> remove uri as it contains internal file storage info
+            if data["storage_class"].is_serializable():
```

Comment:
sorry, should this not be the opposite? if it is serializable that means we *can* put it in the payload, if it is not we remove it?

it really depends on the meaning of the function, WDYT?

Reply:
woops yes lol, will correct

Reply:
nvm all good!